### PR TITLE
Core/cleaning cmake

### PR DESCRIPTION
**Pull request by roigcarlo**
The ones we spoke of this morning + general cleaning

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -487,12 +467,12 @@ sys.path.append(kratos_tests)
...
-    SET(CPACK_GENERATOR "DEB")
-    SET(CPACK_DEBIAN_PACKAGE_MAINTAINER "Kratos Team") #required
+    set(CPACK_GENERATOR "DEB")
+    set(CPACK_DEBIAN_PACKAGE_MAINTAINER "Kratos Team") #required
     set(CPACK_PACKAGE_FILE_NAME  "Kratos_${KratosMultiphysics_VERSION_MAJOR}_${KratosMultiphysics_VERSION_MINOR}")
-    #SET(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
-    SET(CPACK_DEBIAN_PACKAGE_DEPENDS ${CPACK_DEBIAN_PACKAGE_DEPENDS} "libc6, libgcc1, libgfortran3, libgomp1, libopenmpi1.3, libparmetis3.1, libstdc++6, openmpi-bin, ${KRATOS_EXTRA_PACKAGE_DEPENDS} " )
-#    SET(CPACK_DEBIAN_PACKAGE_DEPENDS ${CPACK_DEBIAN_PACKAGE_DEPENDS} " openmpi-bin, libboost-mpi-python-dev, ${KRATOS_EXTRA_PACKAGE_DEPENDS}" )
+    #set(CPACK_DEBIAN_PACKAGE_SHLIBDEPS ON)
+    set(CPACK_DEBIAN_PACKAGE_DEPENDS ${CPACK_DEBIAN_PACKAGE_DEPENDS} "libc6, libgcc1, libgfortran3, libgomp1, libopenmpi1.3, libparmetis3.1, libstdc++6, openmpi-bin, ${KRATOS_EXTRA_PACKAGE_DEPENDS} " )
```
Comment (RiccardoRossi) [inline CMakeLists.txt:474]: this is very old...

**Review by RiccardoRossi — APPROVED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (roigcarlo) [reply CMakeLists.txt:474]: Should remove it then?

Comment (jcotela) [reply CMakeLists.txt:474]: I would be in favor

**Comment by roigcarlo** [comment]
Closing this as we will make several more changes as spoken with the @KratosMultiphysics/technical-committee and I don't want to trigger excessive builds

**Comment by roigcarlo** [comment]
P.s. Do not delete the branch.

**Comment by RiccardoRossi** [comment]
i would add the same "warnings-as-error" that are now in travis (smile @philbucher :-) )

**Comment by philbucher** [comment]
@RiccardoRossi I am not sure if we should do this ...
I.e. it is annoying when you are developing, i.e. the code usually does not right away compile without warnings
=> Devs will delete this from CMake manually I guess

How abt setting this with a CMake-Flag?

sth like `TURN_WARNINGS_INTO_ERRORS` ?
This way the development process would bot be affected and ppl can still try if the compilation is clean before they make a PR

**Comment by roigcarlo** [comment]
I am with @philbucher on this. But we can directly add the flag in the configure if someone want's. I do it sometimes...

Also gl if you try to compile in clang with that flag, we still have a small problem with the components...

**Comment by RiccardoRossi** [comment]
@philbucher @roigcarlo just note that i am not suggesting to turn all warnings into errors, just to activate as errors the one you have in travis. Anyway they need to be fixed before pushing, so i don't think it is a major deal.

anyway...i can obviously live without this change :-)

**Pull request closed without merging**